MyToolsYourToolsBackend/MyToolsYourToolsBackend.Application/Services/RentService.cs

```diff
@@ -30,35 +30,41 @@ public bool CheckIfUserHasEnoughPoints(Guid userId)
             return new PointsModificationToolBorrowingStrategy().Modify(user.Points) > 0;
         }
 
-        public Rent AddRent(RentForCreationDto rent)
+        public Rent AddRent(RentDto rent)
         {
             var rentToSave = Mapper.Map<Rent>(rent);
             var offer = _dbContext.Offers.FirstOrDefault(o => o.Id == rent.OfferId);
             var borrower = _dbContext.Users.FirstOrDefault(u => u.Id == rent.BorrowerId);
             var rentingUser = _dbContext.Users.FirstOrDefault(r => r.Id == offer.OwnerId);
 
             borrower.Rents.Add(rentToSave);
-            offer.Status = Domain.Enums.OfferStatus.Rented;
+            offer.Status = OfferStatus.Rented;
 
             // delete all remaining rentRequests of this offer
-            var rentRequestsToRemove = _dbContext.Notifications
+            IList<Notification> rentRequestsToRemove = _dbContext.Notifications
                 .Where(n => n.OfferId == rent.OfferId
                         && n.Type == NotificationType.RentRequest
-                        && n.TargetUserId != borrower.Id);
-            _dbContext.RemoveRange(rentRequestsToRemove);
+                        && n.TargetUserId != borrower.Id).ToList<Notification>();
 
             _pointsService.ModifyPoints(rentingUser, new PointsModificationToolRentingStrategy());
-
             _pointsService.ModifyPoints(borrower, new PointsModificationToolBorrowingStrategy());
```

Comment:
to to trzeba chyba usunac co nie ?

Reply:
no właśnie to jest o czym napisałem: "Kaucja jest zwracana w momencie usunięcia powiadomienia z bazy (dla zgody i odrzucenia), ale jak jest zgoda to dodanie wypożyczenia w RentService znowu pobiera punkty, także saldo wychodzi jakie powinno."

Reply:
chyba rozumiem